### PR TITLE
chore(vsc): support to init an app when feature flag is enabled

### PR DESCRIPTION
**Pull request by qinezh**
Add support to initialize an existing application when the [feature flag](https://github.com/OfficeDev/TeamsFx/pull/4142) is enabled. 

Please note, this is the draft UX and will be continuously improved according to PM's design.

### CODE REVIEW

File: packages/vscode-extension/package.json [hunk 1]
```diff
@@ -132,7 +136,12 @@
         {
           "id": "teamsfx-empty-project",
           "name": "Teams Toolkit",
-          "when": "fx-extension.sidebarWelcome.default && !fx-extension.canUpgradeToArmAndMultiEnv"
+          "when": "fx-extension.sidebarWelcome.default && !fx-extension.canUpgradeToArmAndMultiEnv && !fx-extension.isInitAppEnabled"
+        },
+        {
+          "id": "teamsfx-empty-project-and-init",
+          "name": "Teams Toolkit",
+          "when": "fx-extension.sidebarWelcome.default && !fx-extension.canUpgradeToArmAndMultiEnv && fx-extension.isInitAppEnabled"
```
Comment (qinezh) [inline packages/vscode-extension/package.json:144]: @HuihuiWu-Microsoft please take a look, the new UX only works when the feature flags enabled and won't affect current UX.

File: packages/fx-core/src/core/FxCore.ts [hunk 2]
```diff
@@ -1419,13 +1416,11 @@ export class FxCore implements v3.ICore {
...
-    return ok(Void);
+    return ok(inputs.projectPath);
   }
+
   @hooks([ErrorHandlerMW, QuestionModelMW, ContextInjectorMW, ProjectSettingsWriterMW])
-  async init(
-    inputs: v2.InputsWithProjectPath,
-    ctx?: CoreHookContext
-  ): Promise<Result<Void, FxError>> {
+  async init(inputs: Inputs, ctx?: CoreHookContext): Promise<Result<string, FxError>> {
```
Comment (qinezh) [inline packages/fx-core/src/core/FxCore.ts:1423]: @jayzhang please take a look at the changes to `init` as I have updates it's method signature in this PR.

File: packages/vscode-extension/src/extension.ts [hunk 3]
```diff
@@ -437,6 +443,12 @@ export async function activate(context: vscode.ExtensionContext) {
     workspacePath && (await isSPFxProject(workspacePath))
   );
 
+  vscode.commands.executeCommand(
+    "setContext",
+    "fx-extension.isInitAppEnabled",
+    workspacePath && isInitAppEnabled()
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/extension.ts:449]: I think we shouldn't include workspacePath check here, right?

File: packages/vscode-extension/package.json (hunk 1 above)
Comment (HuihuiWu-Microsoft) [reply packages/vscode-extension/package.json:144]: LGTM

File: packages/vscode-extension/src/handlers.ts [hunk 4]
```diff
@@ -364,17 +364,30 @@ export async function createNewProjectHandler(args?: any[]): Promise<Result<any,
...
+    await openFolder(result.value, args);
   }
   return result;
 }
 
+export async function initProjectHandler(args?: any[]): Promise<Result<any, FxError>> {
+  ExtTelemetry.sendTelemetryEvent(TelemetryEvent.InitProjectStart, getTriggerFromProperty(args));
+  const result = await runCommand(Stage.init);
+  if (result.isOk()) {
+    await openFolder(result.value, args);
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/handlers.ts:376]: Just to confirm, is readme also included for initialized project?

File: packages/vscode-extension/src/extension.ts (hunk 3 above)
Comment (qinezh) [reply packages/vscode-extension/src/extension.ts:449]: Nice catch! Yes, no need to check workspace here.

File: packages/vscode-extension/src/treeview/treeViewManager.ts [hunk 5]
```diff
@@ -49,6 +49,15 @@ class TreeViewManager {
...
         { name: "new-folder", custom: false }
       ),
+      new TreeViewCommand(
+        StringResources.vsc.commandsTreeViewProvider.initProjectTitleNew,
+        StringResources.vsc.commandsTreeViewProvider.initProjectDescription,
+        "fx-extension.init",
+        vscode.TreeItemCollapsibleState.None,
+        undefined,
+        undefined,
+        { name: "new-folder", custom: false }
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/treeview/treeViewManager.ts:59]: Use the same folder with 'create project'?

File: packages/vscode-extension/src/handlers.ts (hunk 4 above)
Comment (qinezh) [reply packages/vscode-extension/src/handlers.ts:376]: Yes, for current PM spec, `README.md` is included. You could check the spec here https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_git/teamsfx-specs?path=/mods/ga/incremental_dev.md&version=GBjunjiespec&_a=preview&anchor=scaffolded-files

File: packages/vscode-extension/src/treeview/treeViewManager.ts (hunk 5 above)
Comment (qinezh) [reply packages/vscode-extension/src/treeview/treeViewManager.ts:59]: this is the draft UX and will be continuously improved according to PM's design.

File: packages/vscode-extension/src/extension.ts (hunk 3 above)
Comment (qinezh) [reply packages/vscode-extension/src/extension.ts]: Updated, please take a look.

**Review by HuihuiWu-Microsoft — APPROVED** [review]
LGTM

**Review by jayzhang — APPROVED** [review]

**Pull request merged**